lib/absinthe/schema/notation.ex

```diff
@@ -298,6 +298,7 @@ defmodule Absinthe.Schema.Notation do
   @doc false
   # Record an interface type
   def record_interface!(env, identifier, attrs, block) do
+    attrs = Keyword.put(attrs, :identifier, identifier)
```

Comment:
@bruce Note, I've made sure every top level entity now has a top level key `:identifier`. It ends up being reasonably important for user code, and doing `__reference__.identifier` felt dirty.